docker/src/test/java/org/jclouds/docker/compute/SshToCustomPortLiveTest.java

```diff
@@ -0,0 +1,223 @@
+      assertNotNull(listImages);
+      ImageSummary testImage = Iterables.find(listImages, new Predicate<ImageSummary>() {
+         @Override
+         public boolean apply(ImageSummary input) {
+            return imageId.equals(input.id());
+         }
+      });
+
+      assertEquals(testImage.repoTags().size(), 2, "Unexpected number of tags on the image.");
+      assertThat(testImage.repoTags()).contains(toTag(IMAGE_REPOSITORY, IMAGE_TAG_1),
+            toTag(IMAGE_REPOSITORY, IMAGE_TAG_2));
+   }
+
+   /**
+    * Start a node from the newly created image. The dropbear SSH server running
+    * on custom port ( {@value #SSH_PORT}). The Docker networkMode used is
+    * "host". Execute a command through the SSH connection and check the result.
+    * Destroy the node when finished.
+    * 
+    * @throws RunNodesException
+    */
+   @Test(dependsOnMethods = "testImageCreated")
+   public void testCustomPortSsh() throws RunNodesException {
+      final DockerTemplateOptions options = DockerTemplateOptions.Builder
+            .commands("/usr/sbin/dropbear", "-E", "-F", "-p", String.valueOf(SSH_PORT)).overrideLoginUser("root")
+            .overrideLoginPassword("screencast").blockOnPort(SSH_PORT, 30).networkMode("host");
+
+      final Template template = view.getComputeService().templateBuilder().imageId(image.id()).options(options).build();
+
+      String nodeId = null;
+      try {
+         NodeMetadata node = Iterables
+               .getOnlyElement(view.getComputeService().createNodesInGroup("ssh-test", 1, template));
+
+         nodeId = node.getId();
+         ExecResponse response = view.getComputeService().runScriptOnNode(nodeId, "echo hello",
+               runAsRoot(false).wrapInInitScript(false));
+         assertThat(response.getOutput().trim()).endsWith("hello");
+      } finally {
+         if (nodeId != null)
+            view.getComputeService().destroyNode(nodeId);
+      }
+   }
+
+   /**
+    * Build a new image with 2 tags on it in the test preparation phase.
+    * 
+    * @see org.jclouds.apis.BaseContextLiveTest#setupContext()
+    */
+   @Override
+   @BeforeClass(groups = { "integration", "live" })
+   public void setupContext() {
+      super.setupContext();
+
+      final String tag = toTag(IMAGE_REPOSITORY, IMAGE_TAG_1);
+      BuildOptions options = BuildOptions.Builder.tag(tag).verbose(false).nocache(false);
+      InputStream buildImageStream;
+      try {
+         buildImageStream = api().getMiscApi().build(BaseDockerApiLiveTest.tarredDockerfile(), options);
+         consumeStreamSilently(buildImageStream);
+      } catch (IOException e) {
+         e.printStackTrace();
+      }
+      image = api().getImageApi().inspectImage(tag);
+      api().getImageApi().tagImage(image.id(), IMAGE_REPOSITORY, IMAGE_TAG_2, true);
+   }
+
+   /**
+    * After the test remove created image (with all tags).
+    * 
+    * @see #setupContext()
+    */
+   @AfterClass
```

Comment:
Use `@AfterClass(alwaysRun = true)` to make sure the cleanup is executed even if some tests fail.